src/parser/productions.py

```diff
@@ -14,9 +31,17 @@ class PrefixExpression:
     def __init__(self):
         self.op = None
         self.right = None
+        self.folded = False
+
+    def toggle_fold(self):
+        self.folded = not self.folded
```

Comment:
When you click, toggle the folded property. Then i guess you can update the render by calling the `child_node()` method since if folded, that method behaves differently

Reply:
ahhh, siguro for now sige let's keep it. For folding and unfolding in GUI, tkinter has the default logic for it, all we need to do is traverse the tree and insert

Reply:
Do you think the `child_nodes()` method be sufficient for that? Or do you want something different?